libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -575,6 +575,7 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 			for (auto const& arg: arguments)
 			{
 				arg->accept(*this);
+				solAssert(!(arg->annotation().type->category() == Type::Category::Struct), "Struct type cannot be SHA3'd yet.");
```

Comment:
Wouldn't it be better to put that inside `encodeToMemory`? There are other uses of that function.